### PR TITLE
Polymer 3 TS Class: ServerList, ServerCard, ServerConnectionViz

### PR DESCRIPTION
**Pull request by daniellacosse**
<details>
<summary>TODOs</summary>

- [x] TS member order
- [x] fix animation not starting/stopping
- [x] this type issue https://github.com/Jigsaw-Code/outline-client/runs/4243078432?check_suite_focus=true#step:6:18 (irrelevant)
- [x] try to remove LegacyElementMixin (skipping)
- [ ] ugh, this ESModule issue

</details>

### CODE REVIEW

**Comment by daniellacosse** [comment]
HAHA I'm shocked that this passed

File: src/www/ui_components/server_card.ts [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+export enum ServerCardState {
+  CONNECTING = 'CONNECTING',
+  CONNECTED = 'CONNECTED',
+  RECONNECTING = 'RECONNECTING',
+  DISCONNECTING = 'DISCONNECTING',
+  DISCONNECTED = 'DISCONNECTED'
+}
+
+@customElement('server-card') export class ServerCard extends LegacyElementMixin
```
Comment (daniellacosse) [inline src/www/ui_components/server_card.ts:29]: are we cool with `LegacyElementMixin` or should I try to remove it?

**Comment by daniellacosse** [comment]
@fortuna I'm still dealing with the broken ESModule support, but if you get a head start on this I can address your comments while you're out.

File: src/www/ui_components/server_list.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+import {computed, customElement, property} from '@polymer/decorators';
+import {html, PolymerElement} from '@polymer/polymer';
+
+import {Server} from '../model/server';
+
+import {ServerCard} from './server_card';
+
+@customElement('server-list')
+export class ServerList extends PolymerElement {
```
Comment (fortuna) [inline src/www/ui_components/server_list.ts:25]: Let's convert only this element in this PR. The TS code can depend on the JS code (use the .js extension in the import), and even leverage the type guessing from TS. We can also add JSDoc type annotations like we do in the manager.

Example of TS depending on JS: https://github.com/Jigsaw-Code/outline-server/blob/f340c0d484514d306ce5f68a4bb5c7ec82d5a3b5/src/server_manager/web_app/ui_components/outline-gcp-oauth-step.ts

Example js with type annotations depending on both js and ts: https://github.com/Jigsaw-Code/outline-server/blob/f340c0d484514d306ce5f68a4bb5c7ec82d5a3b5/src/server_manager/web_app/ui_components/app-root.js

That was all before Ben converted the elements all to TS: https://github.com/Jigsaw-Code/outline-server/pull/997

**Review by fortuna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**